pkg/utils/managed_service_account_addon.go

```diff
@@ -60,6 +73,71 @@ func IsManagedServiceAccountAddonAvailable(
 	return false
 }
 
+// list globally and get first mch, will return no error and no obj if not found
+func GetMultiClusterHub(
+	hubClient dynamic.Interface,
+) (*unstructured.Unstructured, error) {
+	uMCHList, err := hubClient.Resource(gvrMCH).List(context.TODO(), metav1.ListOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+	if errors.IsNotFound(err) || len(uMCHList.Items) < 1 {
+		// ignore not found errors
+		return nil, nil
+	}
+
+	return &uMCHList.Items[0], nil
+}
+
+// GetMultiClusterEngine find first MCE in the cluster, will return error if no MCE can be found
+func GetMultiClusterEngine(
+	hubClient dynamic.Interface,
+) (*unstructured.Unstructured, error) {
+	uMCEList, err := hubClient.Resource(gvrMCE).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if len(uMCEList.Items) < 1 {
+		// not found will return error
+		return nil, fmt.Errorf("MulticlusterEngine CR not found.")
+	}
+
+	return &uMCEList.Items[0], nil
+}
+
+func EnableManagedServiceAccountFeature(hubClient dynamic.Interface) error {
+	name := ""
+	namespace := ""
+	gvr := gvrMCH
+	patch := fmt.Sprintf(
+		`[{"op":"%s","path":"%s","value":%s}]`,
+		"add", "/spec/componentConfig", `{"managedServiceAccount":{"enable":true}}`,
```

Comment:
has to be this weird format (path & value), otherwise it won't work with omitted version